receiver/k8sclusterreceiver/pods.go

```diff
@@ -0,0 +1,276 @@
+	Description: "Current phase of the pod (1 - Pending, 2 - Running, 3 - Succeeded, 4 - Failed, 5 - Unknown)",
+	Type:        metricspb.MetricDescriptor_GAUGE_INT64,
+}
+
+func getMetricsForPod(pod *corev1.Pod) []*resourceMetrics {
+	metrics := []*metricspb.Metric{
+		{
+			MetricDescriptor: podPhaseMetric,
+			Timeseries: []*metricspb.TimeSeries{
+				utils.GetInt64TimeSeries(int64(phaseToInt(pod.Status.Phase))),
+			},
+		},
+	}
+
+	podRes := getResourceForPod(pod)
+
+	containerResByName := map[string]*resourceMetrics{}
+
+	for _, cs := range pod.Status.ContainerStatuses {
+		if cs.ContainerID == "" {
+			continue
+		}
+
+		contLabels := getAllContainerLabels(cs.ContainerID, cs.Name, cs.Image, podRes.Labels)
+		containerResByName[cs.Name] = &resourceMetrics{resource: getResourceForContainer(contLabels)}
+
+		containerResByName[cs.Name].metrics = getStatusMetricsForContainer(cs)
+	}
+
+	for _, c := range pod.Spec.Containers {
+		cr := containerResByName[c.Name]
+
+		// This likely will not happen since both pod spec and status return
+		// information about the same set of containers. However, if there's
+		// a mismatch, skip collecting spec metrics.
+		if cr == nil {
+			continue
+		}
+
+		cr.metrics = append(cr.metrics, getSpecMetricsForContainer(c)...)
+	}
+
+	out := []*resourceMetrics{
+		{
+			resource: podRes,
+			metrics:  metrics,
+		},
+	}
+
+	out = append(out, listResourceMetrics(containerResByName)...)
+
+	return out
+}
+
+func listResourceMetrics(rms map[string]*resourceMetrics) []*resourceMetrics {
+	out := make([]*resourceMetrics, len(rms))
+
+	i := 0
+	for _, rm := range rms {
+		out[i] = rm
+		i++
+	}
+
+	return out
+}
+
+// getResourceForPod returns a proto representation of the pod.
+func getResourceForPod(pod *corev1.Pod) *resourcepb.Resource {
+	return &resourcepb.Resource{
+		Type: resourcekeys.K8SType,
+		Labels: map[string]string{
+			k8sKeyPodUID:                     string(pod.UID),
+			resourcekeys.K8SKeyPodName:       pod.Name,
+			k8sKeyNodeName:                   pod.Spec.NodeName,
+			resourcekeys.K8SKeyNamespaceName: pod.Namespace,
+			resourcekeys.K8SKeyClusterName:   pod.ClusterName,
+		},
+	}
+}
+
+func phaseToInt(phase corev1.PodPhase) int32 {
+	switch phase {
+	case corev1.PodPending:
+		return 1
+	case corev1.PodRunning:
+		return 2
+	case corev1.PodSucceeded:
+		return 3
+	case corev1.PodFailed:
+		return 4
+	case corev1.PodUnknown:
+		return 5
+	default:
+		return 5
+	}
+}
+
+// getMetadataForPod returns all metadata associated with the pod.
+func getMetadataForPod(pod *corev1.Pod, mc *metadataStore) []*KubernetesMetadata {
+	properties := pod.Labels
+
+	properties[podCreationTime] = pod.CreationTimestamp.Format(time.RFC3339)
+
+	for _, or := range pod.OwnerReferences {
+		properties[strings.ToLower(or.Kind)] = or.Name
+		properties[strings.ToLower(or.Kind)+"_uid"] = string(or.UID)
+
+		// defer syncing replicaset and job workload properties.
+		if or.Kind == "ReplicaSet" || or.Kind == "Job" {
```

Comment:
Can we define consts for `Kind`s?

Reply:
Added constants for `Kind`s in `collector.go`